runtime/include/gpu/rocm/chpl-gpu-gen-includes.h

```diff
@@ -139,6 +139,71 @@ __device__ static inline uint32_t chpl_gpu_getGridDimZ()   {
   return __builtin_amdgcn_grid_size_z() / chpl_gpu_getBlockDimZ();
 }
 
+// =================
+// Atomic Operations
+// =================
+
+#define GPU_2OP_ATOMIC(T, runtime_name, rocm_name)           \
+  __device__ static inline void runtime_name(T *x, T val) {  \
+    rocm_name(x, val);                                       \
+  }                                                          \
+  __host__ static inline void runtime_name(T *x, T val) {}
+
+#define GPU_3OP_ATOMIC(T, runtime_name, rocm_name)                   \
+  __device__ static inline void runtime_name(T *x, T val1, T val2) { \
+    rocm_name(x, val1, val2);                                        \
+  }                                                                  \
+  __host__ static inline void runtime_name(T *x, T val1, T val2) {}
+
+// Some atomic operations are only supported in CUDA while others are only
+// supported in ROCM. We mark the operations that are unsupported for this
+// header's SDK with [*]:
```

Comment:
This is really unfortunate... I was hoping that we could write a common header for these, but I don't think that's possible.

Reply:
I could in theory move the macros to a common header as well as all the invocations of it that are common across APIs.

Reply:
Yeah, but then we would still have some in vendor-specific headers, right? If so, I wouldn't find it as an improvement.

Reply:
Right, we'd have to keep the ones that are CUDA only in the CUDA only header.

On the one hand: less duplicated code, on the other: it's nice to have all these declarations together without having to hop across files and it's nice if we can keep these files more-or-less visually similar so they're easy to compare.

I think my preference is to just keep it "as is", especially given that we don't know how Intel will be with all this.

Reply:
I agree with "as is"